### ISSUE TITLE
Improve Project Navigator file tree

### ISSUE THREAD

**Issue by dabbott** [decosoftware/deco-ide#85]
Issues with the current implementation:
- *a lot* of duct tape to fix performance issues
- confusing communication between desktop and web
- no clearly defined data structures for files and directories
- UI is difficult to style based on the strange component structure, along with reliance on scss

I'm in the process of making a separate npm module for just rendering and controlling a file tree view. Seems like a cohesive, reusable chunk that others would be interested, and more than anything I want it out of our codebase :P

**Comment by noerK** [comment]
I also found a Problem when changing files outside of deco. For example installing node packages :)
Deco don't recognises the changes until you restart deco or reload the whole project.

**Comment by dabbott** [comment]
There's a separate issue for that one https://github.com/decosoftware/deco-ide/issues/61

**Comment by MaximillionMai** [comment]
It would be great to have a filename filter in the project navigator in the left as well.

**Comment by dabbott** [comment]
@MaximillionMai That's an interesting idea... it reminds me of having a layer filter in Sketch. I think I will try it. My only concern is that having another searchbar near the components search in the UI might make it a little confusing which to use. Maybe not.

At some point we'll have a "quick file open" dialog like Atom or Sublime, which will solve most of the same problem, I think.

**Comment by MaximillionMai** [comment]
@dabbott I agree that we should not have two filters right next to each other.  I am wondering why we don't put the filters at the bottom of the list like how XCode does.  the filter for the file/project list is also something from XCode.  It's very useful when say I need to see a list of filtered files and quickly go into all of them to check something one by one.  Quick file open would not work since I would need to look for the current file in the list to find the next one.
